syft/format/common/spdxhelpers/to_format_model.go

```diff
@@ -742,14 +749,28 @@ func toOtherLicenses(catalog *pkg.Collection) []*spdx.OtherLicense {
 	slices.Sort(ids)
 	for _, id := range ids {
 		license := licenses[id]
+		value := license.Value
+		// handle cases where LicenseRef needs to be included in hasExtractedLicensingInfos
+		if license.Value == "" {
+			value, _ = strings.CutPrefix(license.ID, "LicenseRef-")
+		}
 		result = append(result, &spdx.OtherLicense{
 			LicenseIdentifier: license.ID,
-			ExtractedText:     license.Value,
+			ExtractedText:     value,
 		})
 	}
 	return result
 }
 
+// isSingularLicenseRef checks if the string is a singular LicenseRef-* identifier
+func isSingularLicenseRef(s string) bool {
+	// Regular expression to match LicenseRef-* format (case-sensitive)
+	re := regexp.MustCompile(`^LicenseRef-[A-Za-z0-9_-]+$`)
```

Comment:
no need to recompile this regex on every call of this function -- this can be a global like we do elsewhere